// bulma-react-components/dist/components/Button/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _parsePropsAsBulmaClasses = require('../_utils/parsePropsAsBulmaClasses');

var _parsePropsAsBulmaClasses2 = _interopRequireDefault(_parsePropsAsBulmaClasses);

var _utilsfn = require('utilsfn');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Button = function Button(props) {

  var buttonCSSClass = 'button ' + (0, _parsePropsAsBulmaClasses2.default)(props);

  if (props.className) buttonCSSClass = props.className;

  var children = _react2.default.Children.toArray(props.children).map(function (element, index) {
    if ((0, _utilsfn.getTypeOf)(element) === 'String') {
      return _react2.default.createElement(
        'span',
        { key: index },
        element
      );
    } else {
      return element;
    }
  });

  return _react2.default.createElement(
    'button',
    {
      type: props.submit ? 'submit' : 'button',
      'data-spec-selector': props.specSelector,
      className: buttonCSSClass,
      onClick: props.onClick,
      style: props.style,
      disabled: props.disabled || props.isDisabled,
      'aria-hidden': props.isStatic,
      tabIndex: props.isStatic ? -1 : props.tabIndex },
    children
  );
};

Button.propTypes = {
  children: _propTypes2.default.any.isRequired,
  onClick: _propTypes2.default.func,
  disabled: _propTypes2.default.bool
};

exports.default = Button;
