// bulma-react-components/dist/components/FormFields/components/Checkbox/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.propsObject = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Button = require('../../../Button');

var _Button2 = _interopRequireDefault(_Button);

var _Icon = require('../../../Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _Tooltip = require('../../../Tooltip');

var _Tooltip2 = _interopRequireDefault(_Tooltip);

var _propsObject2 = require('./propsObject');

var _propsObject3 = _interopRequireDefault(_propsObject2);

var _parsePropsAsBulmaClasses = require('../../../_utils/parsePropsAsBulmaClasses');

var _parsePropsAsBulmaClasses2 = _interopRequireDefault(_parsePropsAsBulmaClasses);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Checkbox = function Checkbox(props) {
  var checkboxCSSClass = 'checkbox ' + (0, _parsePropsAsBulmaClasses2.default)(props);

  var tooltip = props.info ? _react2.default.createElement(
    _Tooltip2.default,
    { description: props.info },
    '\xA0',
    _react2.default.createElement(
      _Button2.default,
      { isBare: true, tabIndex: -1 },
      _react2.default.createElement(_Icon2.default, { name: 'fas fa-question-circle fa-xs', isInfo: true, isSmall: true })
    )
  ) : null;

  var loading = props.isLoading ? _react2.default.createElement(
    _react2.default.Fragment,
    null,
    '\xA0',
    _react2.default.createElement('span', { className: 'loader' })
  ) : null;

  return _react2.default.createElement(
    'label',
    { htmlFor: props.id || props.name, className: checkboxCSSClass },
    _react2.default.createElement('input', (0, _propsObject3.default)(props)),
    _react2.default.createElement('span', { className: 'checkbox-token' }),
    _react2.default.createElement(
      'span',
      { className: 'checkbox-description' },
      props.description,
      tooltip,
      loading
    )
  );
};

Checkbox.propTypes = {
  id: _propTypes2.default.any,
  name: _propTypes2.default.any,
  value: _propTypes2.default.any,
  defaultValue: _propTypes2.default.any,
  onClick: _propTypes2.default.func,
  onChange: _propTypes2.default.func,
  onInput: _propTypes2.default.func,
  onFocus: _propTypes2.default.func,
  onBlur: _propTypes2.default.func
};

var propsObject = exports.propsObject = _propsObject3.default;

exports.default = Checkbox;
