// bulma-react-components/dist/components/SplashScreen/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

exports.withSplashScreen = withSplashScreen;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTimeout = require('react-timeout');

var _reactTimeout2 = _interopRequireDefault(_reactTimeout);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var defaultOptions = {
  time: 300,
  children: null,
  showSpinner: false,
  top: 0,
  left: 0
};

var SplashScreen = function SplashScreen(props) {
  return _react2.default.createElement(
    'div',
    { className: 'splash-screen' + (props.isVisible ? ' is-visible' : '') + (props.showSpinner ? ' has-spinner' : ''),
      style: props.style },
    props.children
  );
};

function withSplashScreen(Component, options) {
  var SplashScreenWrapper = function (_React$Component) {
    _inherits(SplashScreenWrapper, _React$Component);

    function SplashScreenWrapper(props) {
      _classCallCheck(this, SplashScreenWrapper);

      var _this = _possibleConstructorReturn(this, (SplashScreenWrapper.__proto__ || Object.getPrototypeOf(SplashScreenWrapper)).call(this, props));

      _this.state = { isVisible: true };
      _this.options = _extends({}, defaultOptions, options);
      return _this;
    }

    _createClass(SplashScreenWrapper, [{
      key: 'componentDidMount',
      value: function componentDidMount() {
        var _this2 = this;

        this.props.setTimeout(function () {
          _this2.setState({ isVisible: false });
        }, this.options.time);
      }
    }, {
      key: 'componentWillUnmount',
      value: function componentWillUnmount() {
        this.setState({ isVisible: true });
      }
    }, {
      key: 'render',
      value: function render() {
        return _react2.default.createElement(
          _react2.default.Fragment,
          null,
          _react2.default.createElement(
            SplashScreen,
            {
              isVisible: this.state.isVisible,
              showSpinner: this.options.showSpinner,
              style: { marginTop: this.options.top, marginLeft: this.options.left } },
            this.options.children
          ),
          _react2.default.createElement(Component, this.props)
        );
      }
    }]);

    return SplashScreenWrapper;
  }(_react2.default.Component);

  ;

  SplashScreenWrapper.displayName = 'SplashScreenWrapper(' + getDisplayName(Component) + ')';

  return (0, _reactTimeout2.default)(SplashScreenWrapper);
};

function getDisplayName(Component) {
  return Component.displayName || Component.name || 'Component';
};

exports.default = SplashScreen;
